solvis_graphql_api/composite_solution/composite_rupture_sections.py

```diff
@@ -149,7 +149,7 @@ def build_mfd(
             df["bins"] = pd.cut(df["magnitude"], bins=bins)
             df["bin_center"] = df["bins"].apply(lambda x: x.mid)
             df = df.drop(columns=["magnitude"])
-            df = pd.DataFrame(df.groupby(df.bin_center).sum(numeric_only=True))
+            df = pd.DataFrame(df.groupby(df.bin_center, observed=False).sum(numeric_only=True))
```

Comment:
@DrCuriosity not sure what this is doing, maybe make a comment (maybe with a link to the panda doco) to explain the change?

Reply:
Added documentation on this.

I'm pretty sure `observed=False` is what we still want at this point - certainly, some tests break if it's True.

Given there's a comment on this `build_mfd` function to move it into Solvis at some point, hopefully that provides enough breadcrumbs for future refactoring.